src/persist/src/mem.rs

```diff
@@ -677,13 +676,15 @@ impl BlobMulti for MemBlobMulti {
 /// An in-memory implementation of [Consensus].
 #[derive(Debug)]
 pub struct MemConsensus {
-    data: Arc<TokioMutex<HashMap<String, Vec<VersionedData>>>>,
+    // TODO: This was intended to be a tokio::sync::Mutex but that seems to
```

Comment:
my position at the time was that it requires thinking and justification to use a std Mutex but an async one was definitely correct and we didn't really care about the perf difference. I don't feel strongly but I'm not sure my opinion has changed

at the very least, we should circle back and figure out what's going on in case we want to use this kind of mutex elsewhere